chore: Release v1.7.0

oxarbitrage opened the pull request:
---
name: 'Release Checklist Template'
about: 'Checklist to create and publish a Zebra release'
title: 'Release Zebra (version)'
labels: 'A-release, C-trivial, P-Critical :ambulance:'
assignees: ''

---

# Prepare for the Release

- [x] Make sure there has been [at least one successful full sync test](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=event%3Aschedule) since the last state change, or start a manual full sync.
- [x] Make sure the PRs with the new checkpoint hashes and missed dependencies are already merged.
      (See the release ticket checklist for details)


# Summarise Release Changes

These steps can be done a few days before the release, in the same PR:

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases). We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

To create the final change log:
- [x] Copy the **latest** draft changelog into `CHANGELOG.md` (there can be multiple draft releases)
- [x] Delete any trivial changes
    - [x] Put the list of deleted changelog entries in a PR comment to make reviewing easier
- [x] Combine duplicate changes
- [x] Edit change descriptions so they will make sense to Zebra users
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

## README

README updates can be skipped for urgent releases.

Update the README to:
- [x] Remove any "Known Issues" that have been fixed since the last release.
- [ ] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
- [ ] If Zebra has started using newer Rust language features or standard library APIs, update the known working Rust version in the README, book, and `Cargo.toml`s

You can use a command like:
```sh
fastmod --fixed-strings '1.58' '1.65'
```

## Create the Release PR

- [ ] Push the updated changelog and README into a new branch
      for example: `bump-v1.0.0` - this needs to be different to the tag name
- [x] Create a release PR by adding `&template=release-checklist.md` to the comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/bump-v1.0.0?expand=1&template=release-checklist.md)).
- [x] Freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Mark all the release PRs as `Critical` priority, so they go in the `urgent` Mergify queue.
- [x] Mark all non-release PRs with `do-not-merge`, because Mergify checks approved PRs against every commit, even when a queue is frozen.


# Update Versions and End of Support

## Update Zebra Version

### Choose a Release Level

Zebra follows [semantic versioning](https://semver.org). Semantic versions look like: MAJOR.MINOR.PATCH[-TAG.PRE-RELEASE]

Choose a release level for `zebrad`. Release levels are based on user-visible changes from the changelog:
- Mainnet Network Upgrades are `major` releases
- significant new features or behaviour changes; changes to RPCs, command-line, or configs; and deprecations or removals are `minor` releases
- otherwise, it is a `patch` release

Zebra's Rust API doesn't have any support or stability guarantees, so we keep all the `zebra-*` and `tower-*` crates on a beta `pre-release` version.

### Update Crate Versions

If you're publishing crates for the first time, [log in to crates.io](https://github.com/ZcashFoundation/zebra/blob/doc-crate-own/book/src/dev/crate-owners.md#logging-in-to-cratesio),
and make sure you're a member of owners group.

Check that the release will work:
- [x] Update crate versions, commit the changes to the release branch, and do a release dry-run:

```sh
# Update everything except for alpha crates and zebrad:
cargo release version --verbose --execute --allow-branch '*' --workspace --exclude zebrad --exclude zebra-scan --exclude zebra-grpc beta
# Due to a bug in cargo-release, we need to pass exact versions for alpha crates:
cargo release version --verbose --execute --allow-branch '*' --package zebra-scan 0.1.0-alpha.4
cargo release version --verbose --execute --allow-branch '*' --package zebra-grpc 0.1.0-alpha.2
# Update zebrad:
cargo release version --verbose --execute --allow-branch '*' --package zebrad patch # [ major | minor | patch ]
# Continue with the release process:
cargo release replace --verbose --execute --allow-branch '*' --package zebrad
cargo release commit --verbose --execute --allow-branch '*'
```

Crate publishing is [automatically checked in CI](https://github.com/ZcashFoundation/zebra/actions/workflows/release-crates-io.yml) using "dry run" mode, however due to a bug in `cargo-release` we need to pass exact versions to the alpha crates:

- [x] Update `zebra-scan` and `zebra-grpc` alpha crates in the [release-crates-dry-run workflow script](https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/scripts/release-crates-dry-run.sh)
- [x] Push the above version changes to the release branch.

## Update End of Support

The end of support height is calculated from the current blockchain height:
- [x] Find where the Zcash blockchain tip is now by using a [Zcash explorer](https://zcashblockexplorer.com/blocks) or other tool.
- [x] Replace `ESTIMATED_RELEASE_HEIGHT` in [`end_of_support.rs`](https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/sync/end_of_support.rs) with the height you estimate the release will be tagged.

<details>

<summary>Optional: calculate the release tagging height</summary>

- Add `1152` blocks for each day until the release
- For example, if the release is in 3 days, add `1152 * 3` to the current Mainnet block height

</details>

## Update the Release PR

- [x] Push the version increments and the release constants to the release branch.


# Publish the Zebra Release

## Create the GitHub Pre-Release

- [x] Wait for all the release PRs to be merged
- [x] Create a new release using the draft release as a base, by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0`
- [x] Replace the prepopulated draft changelog in the release description with the final changelog you created;
      starting just _after_ the title `## [Zebra ...` of the current version being released,
      and ending just _before_ the title of the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"
- [x] Delete all the [draft releases from the list of releases](https://github.com/ZcashFoundation/zebra/releases)

## Test the Pre-Release

- [x] Wait until the Docker binaries have been built on `main`, and the quick tests have passed:
    - [x] [ci-unit-tests-docker.yml](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-unit-tests-docker.yml?query=branch%3Amain)
    - [x] [ci-integration-tests-gcp.yml](https://github.com/ZcashFoundation/zebra/actions/workflows/ci-integration-tests-gcp.yml?query=branch%3Amain)
- [x] Wait until the [pre-release deployment machines have successfully launched](https://github.com/ZcashFoundation/zebra/actions/workflows/cd-deploy-nodes-gcp.yml?query=event%3Arelease)

## Publish Release

- [x] [Publish the release to GitHub](https://github.com/ZcashFoundation/zebra/releases) by disabling 'pre-release', then clicking "Set as the latest release"

## Publish Crates

- [x] [Run `cargo login`](https://github.com/ZcashFoundation/zebra/blob/doc-crate-own/book/src/dev/crate-owners.md#logging-in-to-cratesio)
- [ ] Run `cargo clean` in the zebra repo (optional)
- [x] Publish the crates to crates.io: `cargo release publish --verbose --workspace --execute`
- [x] Check that Zebra can be installed from `crates.io`:
      `cargo install --locked --force --version 1.minor.patch zebrad && ~/.cargo/bin/zebrad`
      and put the output in a comment on the PR.

## Publish Docker Images
- [x] Wait for the [the Docker images to be published successfully](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml?query=event%3Arelease).
- [x] Un-freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Remove `do-not-merge` from the PRs you added it to

## Release Failures

If building or running fails after tagging:

<details>

<summary>Tag a new release, following these instructions...</summary>

1. Fix the bug that caused the failure
2. Start a new `patch` release
3. Skip the **Release Preparation**, and start at the **Release Changes** step
4. Update `CHANGELOG.md` with details about the fix
5. Follow the release checklist for the new Zebra version

</details>

oxarbitrage committed:
changelog and readme for 1.7.0 release

oxarbitrage commented:
**Trivial:**

- Bump chrono from 0.4.34 to 0.4.38 (https://github.com/ZcashFoundation/zebra/pull/8400)
- build(deps): bump rustls from 0.21.10 to 0.21.11 in the cargo group (https://github.com/ZcashFoundation/zebra/pull/8418)
- build(deps): bump the devops group with 2 updates (https://github.com/ZcashFoundation/zebra/pull/8457)
- build(deps): bump the devops group with 2 updates (https://github.com/ZcashFoundation/zebra/pull/8408)
- build(deps): bump the prod group with 21 updates (https://github.com/ZcashFoundation/zebra/pull/8402)
- bump(deps): Update what we can from the prod group of deps (https://github.com/ZcashFoundation/zebra/pull/8423)
- change(ci): Updates dependabot.yml (https://github.com/ZcashFoundation/zebra/pull/8433)
- change(test): Disables rejection_restores_internal_state_genesis test on Windows (https://github.com/ZcashFoundation/zebra/pull/8468)
- fix(beta-toolchain): Warnings (https://github.com/ZcashFoundation/zebra/pull/8404)
- fix(dependabot): Typo in dependabot yaml (https://github.com/ZcashFoundation/zebra/pull/8494)
- fix(test): Increases timeout for non-blocking logger test (https://github.com/ZcashFoundation/zebra/pull/8462)
- release(checkpoints): Checkpoints for release 1.7.0 (https://github.com/ZcashFoundation/zebra/pull/8493)
- release(dependencies): Run cargo update (https://github.com/ZcashFoundation/zebra/pull/8498)

oxarbitrage committed:
chore: Release

oxarbitrage committed:
update zebra-scan and zebra-grpc to the release crates dry run script

oxarbitrage committed:
update ESTIMATED_RELEASE_HEIGHT

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ All notable changes to Zebra are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.7.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.7.0) - 2024-05-04
```
```suggestion
## [Zebra 1.7.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.7.0) - 2024-05-06
```

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ All notable changes to Zebra are documented in this file.
+
+### Fixed
+
+- Check Equihash solutions with n=200, k=9 parameters ([#8474](https://github.com/ZcashFoundation/zebra/pull/8474))
```
PR #8474 was part of the Regtest work, so let's move it there.

```suggestion
```

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,33 @@ All notable changes to Zebra are documented in this file.
+### Added
+
+- Preparing for upstream `zcash_client_backend` API changes ([#8425](https://github.com/ZcashFoundation/zebra/pull/8425))
+- Regtest support ([#8383](https://github.com/ZcashFoundation/zebra/pull/8383), [#8421](https://github.com/ZcashFoundation/zebra/pull/8421), [#8368](https://github.com/ZcashFoundation/zebra/pull/8368), [#8413](https://github.com/ZcashFoundation/zebra/pull/8413), [#8475](https://github.com/ZcashFoundation/zebra/pull/8475))
```
```suggestion
- Regtest support ([#8383](https://github.com/ZcashFoundation/zebra/pull/8383), [#8421](https://github.com/ZcashFoundation/zebra/pull/8421), [#8368](https://github.com/ZcashFoundation/zebra/pull/8368), [#8413](https://github.com/ZcashFoundation/zebra/pull/8413), [#8474](https://github.com/ZcashFoundation/zebra/pull/8474), [#8475](https://github.com/ZcashFoundation/zebra/pull/8475))
```

oxarbitrage committed:
update the release period to 6 weeks

oxarbitrage committed:
update release block

oxarbitrage committed:
Apply suggestions from code review

Co-authored-by: Marek <mail@marek.onl>

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,32 @@ All notable changes to Zebra are documented in this file.
 
+## [Zebra 1.7.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.7.0) - 2024-05-06
+
+In this release we introduce Regtest functionality to Zebra and restored Windows support. Also adjusted our Zebra release interval from 2 weeks to 4 weeks approximately.
```
```suggestion
In this release we introduce Regtest functionality to Zebra and restored Windows support. Also adjusted our Zebra release interval from 2 weeks to 6 weeks approximately.
```

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,32 @@ All notable changes to Zebra are documented in this file.
+
+### Changed
+
+- Adjust estimated release interval to once every 4 weeks and the end of support from 16 to 20 weeks ([#8429](https://github.com/ZcashFoundation/zebra/pull/8429))
```
```suggestion
- Adjust estimated release interval to once every 6 weeks and the end of support from 16 to 20 weeks ([#8429](https://github.com/ZcashFoundation/zebra/pull/8429))
```

oxarbitrage committed:
Apply suggestions from code review

Co-authored-by: Marek <mail@marek.onl>

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,32 @@ All notable changes to Zebra are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.7.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.7.0) - 2024-05-06
```
Let's update the date again.

```suggestion
## [Zebra 1.7.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.7.0) - 2024-05-07
```

oxarbitrage committed:
Apply suggestions from code review

Co-authored-by: Marek <mail@marek.onl>

oxarbitrage commented:
```
$ cargo install --locked --force --version 1.7.0 zebrad && ~/.cargo/bin/zebrad
  Downloaded zebrad v1.7.0
  Downloaded 1 crate (260.8 KB) in 1.62s
    Updating crates.io index
  Installing zebrad v1.7.0
    Updating crates.io index
    Updating crates.io index
   Compiling proc-macro2 v1.0.81
   Compiling unicode-ident v1.0.12
   Compiling libc v0.2.154
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling once_cell v1.19.0
   Compiling autocfg v1.2.0
   Compiling serde v1.0.200
   Compiling subtle v2.4.1
   Compiling typenum v1.17.0
   Compiling log v0.4.21
   Compiling pin-project-lite v0.2.14
   Compiling spin v0.5.2
   Compiling memchr v2.7.2
   Compiling ppv-lite86 v0.2.17
   Compiling tracing-core v0.1.32
   Compiling lazy_static v1.4.0
   Compiling nonempty v0.7.0
   Compiling byteorder v1.5.0
   Compiling constant_time_eq v0.3.0
   Compiling generic-array v0.14.7
   Compiling arrayref v0.3.7
   Compiling radium v0.7.0
   Compiling arrayvec v0.7.4
   Compiling memuse v0.2.1
   Compiling crossbeam-utils v0.8.19
   Compiling tap v1.0.1
   Compiling funty v2.0.0
   Compiling thiserror v1.0.59
   Compiling wyz v0.5.1
   Compiling cpufeatures v0.2.12
   Compiling blake2b_simd v1.0.2
   Compiling quote v1.0.36
   Compiling num-traits v0.2.18
   Compiling syn v2.0.60
   Compiling getrandom v0.2.14
   Compiling jobserver v0.1.31
   Compiling bitvec v1.0.1
   Compiling cc v1.0.96
   Compiling rand_core v0.6.4
   Compiling rand_chacha v0.3.1
   Compiling static_assertions v1.1.0
   Compiling either v1.11.0
   Compiling crossbeam-epoch v0.9.18
   Compiling rand v0.8.5
   Compiling num_cpus v1.16.0
   Compiling futures-core v0.3.30
   Compiling futures-sink v0.3.30
   Compiling rayon-core v1.12.1
   Compiling crossbeam-deque v0.8.5
   Compiling semver v1.0.22
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling inout v0.1.3
   Compiling ahash v0.8.11
   Compiling digest v0.10.7
   Compiling zerocopy v0.7.32
   Compiling crunchy v0.2.2
   Compiling slab v0.4.9
   Compiling rayon v1.10.0
   Compiling tinyvec_macros v0.1.1
   Compiling bytes v1.6.0
   Compiling tinyvec v1.6.0
   Compiling universal-hash v0.5.1
   Compiling num-bigint v0.4.4
   Compiling libm v0.2.8
   Compiling opaque-debug v0.3.1
   Compiling bitflags v2.5.0
   Compiling poly1305 v0.8.0
   Compiling aho-corasick v1.1.3
   Compiling num-integer v0.1.46
   Compiling unicode-normalization v0.1.23
   Compiling aead v0.5.2
   Compiling futures-channel v0.3.30
   Compiling mio v0.8.11
   Compiling socket2 v0.5.7
   Compiling signal-hook-registry v1.4.2
   Compiling pin-utils v0.1.0
   Compiling regex-syntax v0.8.3
   Compiling ff v0.13.0
   Compiling futures-task v0.3.30
   Compiling group v0.13.0
   Compiling futures-io v0.3.30
   Compiling pairing v0.23.0
   Compiling pasta_curves v0.5.1
   Compiling bls12_381 v0.8.0
   Compiling secp256k1-sys v0.8.1
   Compiling sha2 v0.10.8
   Compiling incrementalmerkletree v0.5.1
   Compiling jubjub v0.10.0
   Compiling glob v0.3.1
   Compiling halo2_legacy_pdqsort v0.1.0
   Compiling maybe-rayon v0.1.1
   Compiling allocator-api2 v0.2.18
   Compiling regex-automata v0.4.6
   Compiling itoa v1.0.11
   Compiling litrs v0.4.1
   Compiling hashbrown v0.14.5
   Compiling document-features v0.2.8
   Compiling clang-sys v1.7.0
   Compiling ring v0.16.20
   Compiling rustix v0.38.34
   Compiling syn v1.0.109
   Compiling base64ct v1.0.1
   Compiling prettyplease v0.2.19
   Compiling password-hash v0.3.2
   Compiling serde_derive v1.0.200
   Compiling tracing-attributes v0.1.27
   Compiling zeroize_derive v1.4.2
   Compiling thiserror-impl v1.0.59
   Compiling tokio-macros v2.2.0
   Compiling zeroize v1.7.0
   Compiling cipher v0.4.4
   Compiling futures-macro v0.3.30
   Compiling chacha20 v0.9.1
   Compiling chacha20poly1305 v0.10.1
   Compiling cbc v0.1.2
   Compiling fpe v0.6.1
   Compiling zcash_note_encryption v0.4.0
   Compiling aes v0.8.4
   Compiling tracing v0.1.40
   Compiling tokio v1.37.0
   Compiling halo2_proofs v0.3.0
   Compiling regex v1.10.4
   Compiling futures-util v0.3.30
   Compiling blake2s_simd v1.0.2
   Compiling equivalent v1.0.1
   Compiling untrusted v0.7.1
   Compiling minimal-lexical v0.2.1
   Compiling ident_case v1.0.1
   Compiling smallvec v1.13.2
   Compiling linux-raw-sys v0.4.13
   Compiling nom v7.1.3
   Compiling pbkdf2 v0.10.1
   Compiling zcash_protocol v0.1.1
   Compiling bs58 v0.5.1
   Compiling libloading v0.8.3
   Compiling hmac v0.12.1
   Compiling f4jumble v0.1.0
   Compiling zcash_encoding v0.2.0
   Compiling bindgen v0.69.4
   Compiling bech32 v0.9.1
   Compiling home v0.5.9
   Compiling which v4.4.2
   Compiling hex v0.4.3
   Compiling indexmap v2.2.6
   Compiling rustc_version v0.4.0
   Compiling secp256k1 v0.26.0
   Compiling uint v0.9.5
   Compiling reddsa v0.5.1
   Compiling cexpr v0.6.0
   Compiling halo2_gadgets v0.3.0
   Compiling zcash_address v0.3.2
   Compiling bip0039 v0.10.1
   Compiling redjubjub v0.7.0
   Compiling futures-executor v0.3.30
   Compiling itertools v0.12.1
   Compiling ripemd v0.1.3
   Compiling hdwallet v0.4.1
   Compiling equihash v0.2.0
   Compiling lock_api v0.4.12
   Compiling serde_json v1.0.116
   Compiling lazycell v1.3.0
   Compiling fnv v1.0.7
   Compiling shlex v1.3.0
   Compiling strsim v0.10.0
   Compiling platforms v3.4.0
   Compiling rustc-hash v1.1.0
   Compiling darling_core v0.20.8
   Compiling futures v0.3.30
   Compiling curve25519-dalek v4.1.2
   Compiling pin-project-internal v1.1.5
   Compiling scopeguard v1.2.0
   Compiling ryu v1.0.17
   Compiling tokio-util v0.7.10
   Compiling pin-project v1.1.5
   Compiling curve25519-dalek-derive v0.1.1
   Compiling crossbeam-channel v0.5.12
   Compiling rustc-hex v2.1.0
   Compiling pkg-config v0.3.30
   Compiling fixed-hash v0.8.0
   Compiling link-cplusplus v1.0.9
   Compiling vcpkg v0.2.15
   Compiling signature v2.2.0
   Compiling tower-service v0.3.2
   Compiling getrandom v0.1.16
   Compiling ed25519 v2.2.3
   Compiling primitive-types v0.12.2
   Compiling darling_macro v0.20.8
   Compiling bellman v0.14.0
   Compiling bridgetree v0.4.0
   Compiling lz4-sys v1.9.4
   Compiling zip32 v0.1.1
   Compiling libz-sys v1.1.16
   Compiling zcash_spec v0.1.0
   Compiling parking_lot_core v0.9.10
   Compiling orchard v0.6.0
   Compiling iana-time-zone v0.1.60
   Compiling termcolor v1.4.1
   Compiling darling v0.20.8
   Compiling serde_with_macros v3.8.1
   Compiling cxxbridge-flags v1.0.113
   Compiling humantime v2.1.0
   Compiling regex-syntax v0.6.29
   Compiling portable-atomic v1.6.0
   Compiling unicode-width v0.1.12
   Compiling codespan-reporting v0.11.1
   Compiling cxx v1.0.113
   Compiling ed25519-zebra v4.0.3
   Compiling x25519-dalek v2.0.1
   Compiling zcash_history v0.4.0
   Compiling chrono v0.4.38
   Compiling zcash_primitives v0.13.0
   Compiling serde_with v3.8.1
   Compiling regex-automata v0.1.10
   Compiling bitflags-serde-legacy v0.1.1
   Compiling serde-big-array v0.5.1
   Compiling displaydoc v0.2.4
   Compiling hdrhistogram v7.5.4
   Compiling overload v0.1.1
   Compiling tower-layer v0.3.2
   Compiling rustversion v1.0.15
   Compiling console v0.15.8
   Compiling tower v0.4.13
   Compiling nu-ansi-term v0.46.0
   Compiling rand_core v0.5.1
   Compiling matchers v0.1.0
   Compiling zebra-chain v1.0.0-beta.37
   Compiling librocksdb-sys v0.16.0+8.10.0
   Compiling cxx-gen v0.7.121
   Compiling sapling-crypto v0.1.3
   Compiling orchard v0.7.1
   Compiling libgit2-sys v0.16.2+1.7.2
   Compiling spin v0.9.8
   Compiling nanorand v0.7.0
   Compiling tracing-log v0.2.0
   Compiling sharded-slab v0.1.7
   Compiling thread_local v1.1.8
   Compiling unicode-bidi v0.3.15
   Compiling xdg v2.5.2
   Compiling camino v1.1.6
   Compiling known-folders v1.1.0
   Compiling option-ext v0.2.0
   Compiling number_prefix v0.4.0
   Compiling percent-encoding v2.3.1
   Compiling form_urlencoded v1.2.1
   Compiling indicatif v0.17.8
   Compiling dirs-sys v0.4.1
   Compiling idna v0.5.0
   Compiling zcash_script v0.1.16
   Compiling tracing-subscriber v0.3.18
   Compiling zcash_primitives v0.14.0
   Compiling flume v0.10.14
   Compiling parking_lot v0.12.2
   Compiling rand_chacha v0.2.2
   Compiling cxxbridge-macro v1.0.113
   Compiling metrics-macros v0.7.1
   Compiling unicase v2.7.0
   Compiling anyhow v1.0.82
   Compiling rlimit v0.10.1
   Compiling httparse v1.8.0
   Compiling fastrand v2.1.0
   Compiling utf8parse v0.2.1
   Compiling parking_lot_core v0.8.6
   Compiling powerfmt v0.2.0
   Compiling deranged v0.3.11
   Compiling http v0.2.12
   Compiling tempfile v3.10.1
   Compiling anstyle-parse v0.2.4
   Compiling metrics v0.21.1
   Compiling vergen v8.3.1
   Compiling rand v0.7.3
   Compiling zcash_proofs v0.14.0
   Compiling howudoin v0.1.2
   Compiling url v2.5.0
   Compiling dirs v5.0.1
   Compiling humantime-serde v1.1.1
   Compiling tracing-futures v0.2.5
   Compiling tokio-stream v0.1.15
   Compiling env_logger v0.7.1
   Compiling cargo-platform v0.1.8
   Compiling backtrace v0.3.71
   Compiling metrics v0.22.3
   Compiling bstr v1.9.1
   Compiling instant v0.1.12
   Compiling adler v1.0.2
   Compiling eyre v0.6.12
   Compiling anstyle-query v1.0.3
   Compiling is_terminal_polyfill v1.70.0
   Compiling gimli v0.28.1
   Compiling num-conv v0.1.0
   Compiling num_threads v0.1.7
   Compiling time-core v0.1.2
   Compiling try-lock v0.2.5
   Compiling anstyle v1.0.7
   Compiling colorchoice v1.0.1
   Compiling want v0.3.1
   Compiling anstream v0.6.14
   Compiling time v0.3.36
   Compiling globset v0.4.14
   Compiling miniz_oxide v0.7.2
   Compiling addr2line v0.21.0
   Compiling cargo_metadata v0.18.1
   Compiling quickcheck v0.9.2
   Compiling http-body v0.4.6
   Compiling quickcheck_macros v0.9.1
   Compiling jsonrpc-core v18.0.0
   Compiling bincode v1.3.3
   Compiling toml v0.5.11
   Compiling tokio-util v0.6.10
   Compiling object v0.32.2
   Compiling fs-err v2.11.0
   Compiling rustc-demangle v0.1.23
   Compiling mset v0.1.1
   Compiling wagyu-zcash-parameters-3 v0.2.0
   Compiling cfg-if v0.1.10
   Compiling hex-literal v0.4.1
   Compiling unicode-xid v0.2.4
   Compiling wagyu-zcash-parameters-2 v0.2.0
   Compiling httpdate v1.0.3
   Compiling human_bytes v0.4.3
   Compiling wagyu-zcash-parameters-4 v0.2.0
   Compiling wagyu-zcash-parameters-1 v0.2.0
   Compiling heck v0.5.0
   Compiling strsim v0.11.1
   Compiling wagyu-zcash-parameters-5 v0.2.0
   Compiling clap_lex v0.7.0
   Compiling indenter v0.3.3
   Compiling wagyu-zcash-parameters-6 v0.2.0
   Compiling wagyu-zcash-parameters v0.2.0
   Compiling clap_builder v4.5.2
   Compiling clap_derive v4.5.4
   Compiling hyper v0.14.28
   Compiling synstructure v0.12.6
   Compiling net2 v0.2.39
   Compiling proc-macro-crate v0.1.5
   Compiling jsonrpc-server-utils v18.0.0
   Compiling ordered-map v0.4.2
   Compiling parking_lot v0.11.2
   Compiling tower-batch-control v0.2.41-beta.13
   Compiling tracing-error v0.2.0
   Compiling zebra-node-services v1.0.0-beta.37
   Compiling zcash_proofs v0.13.0
   Compiling tower-fallback v0.2.41-beta.13
   Compiling serde_spanned v0.6.5
   Compiling toml_datetime v0.6.5
   Compiling owo-colors v3.5.0
   Compiling winnow v0.6.7
   Compiling color-eyre v0.6.3
   Compiling zebra-network v1.0.0-beta.37
   Compiling toml_edit v0.22.12
   Compiling clap v4.5.4
   Compiling jsonrpc-http-server v18.0.0
   Compiling jsonrpc-derive v18.0.0
   Compiling abscissa_derive v0.7.0
   Compiling secrecy v0.8.0
   Compiling wait-timeout v0.2.0
   Compiling tracing-log v0.1.4
   Compiling canonical-path v2.0.2
   Compiling arc-swap v1.7.1
   Compiling toml v0.8.12
   Compiling abscissa_core v0.7.0
   Compiling tracing-appender v0.2.3
   Compiling atty v0.2.14
   Compiling git2 v0.18.3
   Compiling zebra-script v1.0.0-beta.37
   Compiling zebrad v1.7.0
   Compiling rocksdb v0.22.0
   Compiling zebra-state v1.0.0-beta.37
   Compiling zebra-consensus v1.0.0-beta.37
   Compiling zebra-rpc v1.0.0-beta.37
    Finished release [optimized] target(s) in 4m 37s
   Replacing /home/alfredo/.cargo/bin/zebrad
    Replaced package `zebrad v1.5.2` with `zebrad v1.7.0` (executable `zebrad`)























































              X@8:::::::@X                      S888@t             S888%        
          @::X:;SX;%8S8@@:X::8              S:         .8      @.         .@    
       @:;@8@.S8  ;;t;. XX.;@@;:@         ;               %  ;               X  
     8:@88     .%@.S@.XS:     888:8      8                 t%                 @ 
    :;SX    XS8;::@X@8::;8t8:   8;;:@                                           
  @::@    88:::8 8::88::::::X@    8::8  @                                      8
 @:XS8   ;::::::8.88%8@S88::::8:  8S8:8 X                                      X
 ::;8 t8S8::::@St 88SX @SS@8:8S;8S @@:: 8                                      8
8:XX  %%@@::X% 8:  X@ t:8t X ..%t8  ;8:8                                        
::8t:  tX:X8tS 8: :@ ;.8@.X tt888:  @8:: X                                    % 
::8S:  @:8Xtt% 8: S 8 8S.   .8;::.  88::  %                                  t  
::X8   S@8   8:8.@8@8Xt S  .%8;S    %8::   8                                8   
 ::@8  S .   @:::::::X;@ 8.88S8    @8;:X     %                            %     
 8:X8X .@:X88:::::::::888 @;:X    %88::        %                        %       
  @::S    888@8:::::::::8X; 8     8::8           8                    8         
   X:;S@   tXX8;::::::::S::@8   8;S:8              S.              :t           
     ::8S@     :S88XX@8X:     @XX::                   %          %              
       8:;%X8.:S        t; X8%S:8X                      @      S                
         X8::8SS8SX8@@X@8;8S:8X                           S  %                  
              X@8:::::::@XS                                %                    

Thank you for running a mainnet zebrad 1.7.0 node!
You're helping to strengthen the network and contributing to a social good :)
2024-05-07T17:30:31.686133Z  INFO zebrad::components::tracing::component: started tracing component filter="info" TRACING_STATIC_MAX_LEVEL=LevelFilter::INFO LOG_STATIC_MAX_LEVEL=Info
2024-05-07T17:30:31.686588Z  INFO zebrad::components::tracing::component: set 'tracing.progress_bar ="summary"' in zebrad.toml to activate progress bars
2024-05-07T17:30:31.686628Z  INFO zebrad::application: Diagnostic Metadata:
version: 1.7.0
Zcash network: Mainnet
running state version: 25.3.0
initial disk state version: 25.3.0
features: default,getblocktemplate_rpcs,howudoin,indicatif,progress_bar,release_max_level_info
target triple: x86_64-unknown-linux-gnu
rust compiler: 1.77.2
rust release date: 2024-04-09
optimization level: 3
debug checks: false
2024-05-07T17:30:31.686671Z  INFO zebrad::application: loaded zebrad config config_path=None config=ZebradConfig { consensus: Config { checkpoint_sync: true }, metrics: Config { endpoint_addr: None }, network: Config { listen_addr: 0.0.0.0:8233, external_addr: None, network: Mainnet, initial_mainnet_peers: {"dnsseed.z.cash:8233", "dnsseed.str4d.xyz:8233", "mainnet.seeder.zfnd.org:8233", "mainnet.is.yolo.money:8233"}, initial_testnet_peers: {"dnsseed.testnet.z.cash:18233", "testnet.seeder.zfnd.org:18233", "testnet.is.yolo.money:18233"}, cache_dir: IsEnabled(true), peerset_initial_target_size: 25, crawl_new_peer_interval: 61s, max_connections_per_ip: 1 }, state: Config { cache_dir: "/home/alfredo/.cache/zebra", ephemeral: false, delete_old_database: true, debug_stop_at_height: None, debug_validity_check_interval: None }, tracing: Config { inner: InnerConfig { use_color: true, force_use_color: false, filter: None, buffer_limit: 128000, endpoint_addr: None, flamegraph: None, progress_bar: None, log_file: None, use_journald: false } }, sync: Config { download_concurrency_limit: 50, checkpoint_verify_concurrency_limit: 1000, full_verify_concurrency_limit: 20, parallel_cpu_threads: 0 }, mempool: Config { tx_cost_limit: 80000000, eviction_memory_time: 3600s, debug_enable_at_height: None }, rpc: Config { listen_addr: None, parallel_cpu_threads: 0, debug_force_finished_sync: false }, mining: Config { miner_address: None, extra_coinbase_data: None, debug_like_zcashd: true } }
2024-05-07T17:30:31.687810Z  INFO {net="Main"}: zebrad::application: initialized rayon thread pool for CPU-bound tasks num_threads=12
2024-05-07T17:30:31.691772Z  INFO {net="Main"}: zebrad::commands::start: Starting zebrad
2024-05-07T17:30:31.692071Z  INFO {net="Main"}: zebrad::commands::start: initializing node state
2024-05-07T17:30:31.706059Z  INFO {net="Main"}: zebrad::commands::start: opening database, this may take a few minutes
2024-05-07T17:30:31.706435Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: trying to open current database format running_version=25.3.0
2024-05-07T17:30:31.707205Z  INFO zebra_state::service::finalized_state::disk_db: the open file limit is high enough for Zebra current_limit=1024 min_limit=512 ideal_limit=1024
2024-05-07T17:30:32.923291Z  INFO zebra_state::service::finalized_state::disk_db: Opened Zebra state cache at /home/alfredo/.cache/zebra/state/v25/mainnet
2024-05-07T17:30:32.924508Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: checking database format produced by a previous zebra instance is current and valid running_version=25.3.0
2024-05-07T17:30:32.924462Z  INFO zebra_state::service::finalized_state: loaded Zebra state cache tip=Some((Height(449981), block::Hash("0000000000d187172298564d866cd7e2ee2b946ec524872fcd4b2afda2b2d84e")))
2024-05-07T17:30:32.926389Z  INFO zebra_state::service: starting legacy chain check
2024-05-07T17:30:32.937907Z  INFO zebra_state::service: cached state consensus branch is valid: no legacy chain found
2024-05-07T17:30:32.937986Z  INFO {net="Main"}: zebrad::commands::start: logging database metrics on startup
2024-05-07T17:30:32.938020Z  INFO {net="Main"}: zebra_state::service::finalized_state::disk_db: the open file limit is high enough for Zebra current_limit=1024 min_limit=512 ideal_limit=1024
2024-05-07T17:30:32.940020Z  INFO {net="Main"}: zebra_state::service::finalized_state::disk_db: Total Database Disk Size: 37.6 GB
2024-05-07T17:30:32.940032Z  INFO {net="Main"}: zebra_state::service::finalized_state::disk_db: Total Live Data Disk Size: 18.7 GB
2024-05-07T17:30:32.940037Z  INFO {net="Main"}: zebra_state::service::finalized_state::disk_db: Total Database Memory Size: 49.2 KB
2024-05-07T17:30:32.940164Z  INFO {net="Main"}: zebrad::commands::start: initializing network
2024-05-07T17:30:32.940269Z  INFO {net="Main"}:open_listener{addr=0.0.0.0:8233}: zebra_network::peer_set::initialize: Trying to open Zcash protocol endpoint at 0.0.0.0:8233...
2024-05-07T17:30:32.940341Z  INFO {net="Main"}:open_listener{addr=0.0.0.0:8233}: zebra_network::peer_set::initialize: Opened Zcash protocol endpoint at 0.0.0.0:8233
2024-05-07T17:30:32.940589Z  INFO {net="Main"}: zebra_network::address_book_updater: starting the address book updater
2024-05-07T17:30:33.188252Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="mainnet.seeder.zfnd.org:8233" remote_ip_count=25
2024-05-07T17:30:33.200251Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="dnsseed.z.cash:8233" remote_ip_count=25
2024-05-07T17:30:33.249608Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: resolved seed peer IP addresses seed="dnsseed.str4d.xyz:8233" remote_ip_count=29
2024-05-07T17:30:33.813889Z  INFO zebra_state::service::finalized_state::disk_format::upgrade: database format is valid running_version=25.3.0 inital_disk_version=25.3.0
2024-05-07T17:30:37.941217Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: DNS timeout resolving peer IP addresses host="mainnet.is.yolo.money:8233" e=Elapsed(())
2024-05-07T17:30:37.941285Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: Seed peer DNS resolution failed, checking for addresses from other seed peers host="mainnet.is.yolo.money:8233" attempts=1
2024-05-07T17:30:37.944830Z  INFO {net="Main"}:add_initial_peers: zebra_network::config: loaded cached peer IP addresses cached_ip_count=24 peer_cache_file="/home/alfredo/.cache/zebra/network/mainnet.peers"
2024-05-07T17:30:37.944906Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: limiting the initial peers list from 66 to 25
2024-05-07T17:30:37.945183Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: connecting to initial peer set ipv4_peer_count=22 ipv6_peer_count=3
2024-05-07T17:30:39.085155Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: an initial peer connection failed successes=5 errors=1 addr=v6redacted:8233 e=No route to host (os error 113)
2024-05-07T17:30:39.259293Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: an initial peer connection failed successes=6 errors=2 addr=v4redacted:8233 e=Serialization error
2024-05-07T17:30:40.572742Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: an initial peer connection failed successes=19 errors=3 addr=v4redacted:8233 e=Connection refused (os error 111)
2024-05-07T17:30:41.272610Z  INFO {net="Main"}:add_initial_peers: zebra_network::peer_set::initialize: finished connecting to initial seed and disk cache peers handshake_success_total=22 handshake_error_total=3 outbound_connections=22
2024-05-07T17:30:41.273133Z  INFO {net="Main"}: zebra_network::peer_set::initialize: sending initial request for peers active_initial_peer_count=22
2024-05-07T17:30:41.275980Z  INFO {net="Main"}: zebrad::commands::start: initializing verifiers
2024-05-07T17:30:41.276177Z  INFO {net="Main"}:crawl_and_dial{new_peer_interval=61s}: zebra_network::peer_set::initialize: starting the peer address crawler crawl_new_peer_interval=61s outbound_connections=22
2024-05-07T17:30:41.276430Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true } network=Mainnet}: zebra_consensus::router: starting state checkpoint validation
2024-05-07T17:30:41.299003Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true } network=Mainnet}: zebra_consensus::router: initializing block verifier router tip=Some((Height(449981), block::Hash("0000000000d187172298564d866cd7e2ee2b946ec524872fcd4b2afda2b2d84e"))) max_checkpoint_height=Height(2491606)
2024-05-07T17:30:41.299562Z  INFO {net="Main"}: zebrad::commands::start: initializing syncer
2024-05-07T17:30:41.299763Z  INFO {net="Main"}: zebrad::commands::start: initializing mempool
2024-05-07T17:30:41.299940Z  INFO {net="Main"}: zebrad::commands::start: fully initializing inbound peer request handler
2024-05-07T17:30:41.299962Z  INFO {net="Main"}: zebrad::commands::start: spawning RPC server
2024-05-07T17:30:41.300197Z  INFO {net="Main"}: zebrad::commands::start: spawning block gossip task
2024-05-07T17:30:41.300243Z  INFO {net="Main"}: zebrad::commands::start: spawning mempool queue checker task
2024-05-07T17:30:41.300262Z  INFO {net="Main"}: zebrad::commands::start: spawning mempool transaction gossip task
2024-05-07T17:30:41.300281Z  INFO {net="Main"}: zebrad::commands::start: spawning delete old databases task
2024-05-07T17:30:41.300319Z  INFO {net="Main"}: zebrad::components::sync::gossip: initializing block gossip task
2024-05-07T17:30:41.300348Z  INFO {net="Main"}: zebrad::commands::start: spawning progress logging task
2024-05-07T17:30:41.300337Z  INFO {net="Main"}: zebrad::components::mempool::queue_checker: initializing mempool queue checker task
2024-05-07T17:30:41.300401Z  INFO {net="Main"}: zebrad::commands::start: spawning end of support checking task
2024-05-07T17:30:41.300436Z  INFO {net="Main"}: zebrad::commands::start: spawning mempool crawler task
2024-05-07T17:30:41.300435Z  INFO {net="Main"}: zebrad::components::mempool::gossip: initializing transaction gossip task
2024-05-07T17:30:41.300429Z  INFO {net="Main"}: zebra_state::config: checking for old database versions db_kind="state"
2024-05-07T17:30:41.300458Z  INFO {net="Main"}: zebrad::commands::start: spawning syncer task
2024-05-07T17:30:41.300462Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Starting end of support task
2024-05-07T17:30:41.300529Z  INFO {net="Main"}: zebrad::commands::start: spawned initial Zebra tasks
2024-05-07T17:30:41.300583Z  INFO {net="Main"}: zebrad::components::mempool::crawler: initializing mempool crawler task
2024-05-07T17:30:41.300709Z  INFO {net="Main"}: zebra_state::config: finished old database version cleanup task
2024-05-07T17:30:41.358154Z  INFO {net="Main"}:sync:try_to_sync: zebrad::components::sync: starting sync, obtaining new tips state_tip=Some(Height(449981))
2024-05-07T17:30:42.369084Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true } network=Mainnet}: zebra_consensus::router: state is not fully synced yet, remaining checkpoints will be verified during syncing
2024-05-07T17:30:42.369097Z  INFO {net="Main"}:init{config=Config { checkpoint_sync: true } network=Mainnet}: zebra_consensus::router: finished state checkpoint validation
2024-05-07T17:30:46.880402Z  INFO {net="Main"}:sync:try_to_sync:try_to_sync_once{extra_hashes={}}: zebrad::components::sync: extending tips tips.len=1 in_flight=499 extra_hashes=0 lookahead_limit=1000 state_tip=Some(Height(449981))
2024-05-07T17:30:50.690764Z  INFO {net="Main"}:sync:try_to_sync:try_to_sync_once{extra_hashes={}}: zebrad::components::sync: extending tips tips.len=1 in_flight=997 extra_hashes=0 lookahead_limit=1000 state_tip=Some(Height(449981))
2024-05-07T17:30:51.301890Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Checking if Zebra release is inside support range ...
2024-05-07T17:30:51.301902Z  INFO {net="Main"}: zebrad::components::sync::end_of_support: Zebra release is supported until block 2576762, please report bugs at https://github.com/ZcashFoundation/zebra/issues
2024-05-07T17:31:01.278084Z  INFO {net="Main"}:peer_cache_updater: zebra_network::config: updated cached peer IP addresses cached_ip_count=52 peer_cache_file="/home/alfredo/.cache/zebra/network/mainnet.peers"
2024-05-07T17:31:04.832807Z  INFO {net="Main"}:sync:try_to_sync:obtain_tips:checkpoint: zebra_consensus::checkpoint: verified checkpoint range block_count=1200 current_range=(Excluded(Height(449981)), Included(Height(451181)))
...

```
